src/ln/peer_handler.rs

```diff
@@ -348,6 +351,22 @@ impl<Descriptor: SocketDescriptor> PeerManager<Descriptor> {
 							peer.pending_read_buffer_pos = 0;
 
 							macro_rules! encode_and_send_msg {
+								($msg: expr, $msg_code: expr) => {
+									{
+										encode_and_send_actual_msg!($msg, $msg_code);
+										if peer.last_synced_channel > -1{
+											let all_messages_tuple = self.message_handler.route_handler.get_next_announcements(peer.last_synced_channel,5);
```

Comment:
This looks about right, but its in the wrong place - we dont want to keep sending when they send us data for us to process that results in us sending response data, but instead when our outbound buffer gets drained and the rust-lightning user calls write_event. At that point we should call do_attempt_write_data (which it does) and then check how many messages are in the queue and fill the queue all the way up (I think current limit is 10 in read_event, though that should be const-ified) with get_next_announcement.